File: programming_language_bindings/VisualBasic/Examples/FullDemo/AboutDlg.vb
Imports System.Windows.Forms

Public Class AboutDlg

    Private Sub OK_Button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs)
        Me.DialogResult = System.Windows.Forms.DialogResult.OK
        Me.Close()
    End Sub

    Private Sub Cancel_Button_Click(ByVal sender As System.Object, ByVal e As System.EventArgs)
        Me.DialogResult = System.Windows.Forms.DialogResult.Cancel
        Me.Close()
    End Sub

    Private Sub AboutDlg_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Dim nChars As Integer = DTWAINAPI.DTWAIN_GetVersionInfo(IntPtr.Zero, -1)
        Dim szInfo As New System.Text.StringBuilder(nChars)
        DTWAINAPI.DTWAIN_GetVersionInfo(szInfo, nChars)
        edInfo.Text = szInfo.ToString()
    End Sub
End Class

File: demos/csharp/Examples/FullDemo/AboutDlg.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Dynarithmic;

namespace TWAINDemo
{
    public partial class AboutDlg : Form
    {
        public AboutDlg()
        {
            InitializeComponent();
        }

        private void AboutDlg_Load(object sender, EventArgs e)
        {
            int nChars = TwainAPI.DTWAIN_GetVersionInfo(IntPtr.Zero, -1);
            StringBuilder szInfo = new StringBuilder(nChars);
            TwainAPI.DTWAIN_GetVersionInfo(szInfo, nChars);
            edInfo.Text = szInfo.ToString().Replace("\n", "\r\n");
        }
    }
}
